src/redis-cli.c

```diff
@@ -2209,19 +2222,42 @@ static void repl(void) {
 
 static int noninteractive(int argc, char **argv) {
     int retval = 0;
+    sds stdin_tag_name = NULL;
     sds *sds_args = getSdsArrayFromArgv(argc, argv, config.quoted_input);
+
     if (!sds_args) {
         printf("Invalid quoted string\n");
         return 1;
     }
+
     if (config.stdinarg) {
         sds_args = sds_realloc(sds_args, (argc + 1) * sizeof(sds));
         sds_args[argc] = readArgFromStdin();
         argc++;
+    } else if (config.stdin_tag_arg) {
+        int i = 0, tag_match = 0;
+        stdin_tag_name = sdsnew(config.stdin_tag_name);
+
+        for (; i < argc; i++) {
+            if (sdscmp(stdin_tag_name, sds_args[i]) != 0) continue;
```

Comment:
i think this can be a plain `strcmp` and we don't need to allocate (convert) the `stdin_tag_name` to an sds.
command line args won't have null charts in them.